alignment_distribution.Rmd

```diff
@@ -0,0 +1,121 @@
+---
+title: "Party Alignment Distribution"
+output: html_document
+---
+
+```{r setup, include=FALSE}
+library(mixtools)
+
+load("./data/Global Party Survey by Party Stata V2_1_Apr_2020.RData")
+table <- data$add_influence_column(table)
+```
+
+## Alignment Distributions
+
+Let's look at the ideological alignments of all political parties globally and see if they adhere to some distribution. We'll be looking specifically at the economic alignments of the parties along a left-right axis, as reported by local political experts.
+
+## Without Influence Weighting
+
+A priori, having not looked at the data, it might be reasonable to assume that the counts follow a normal distribution. In other words, we might assume that political parties tend to tack to the center (especially after weighting for influence) and that there are far fewer parties to either extreme.
+
+Let's test this naive assumption. We can calculate the mean and standard deviation from the data and overlay it on our histogram:
+
+```{r}
+mu <- mean(table$V4_Scale, na.rm = TRUE)
+sigma <- sd(table$V4_Scale, na.rm = TRUE)
+
+hist(table$V4_Scale, probability = TRUE)
+curve(dnorm(x, mean = mu, sd = sigma), add = TRUE)
+```
+
+We can pretty clearly see from the graphs that the naive assumption is incorrect, but let's test that hypothesis. We can calculate _expected_ counts for each of our ten bins by integrating over the distribution. Then we can run a chi-square test to see if our _observed_ bins might likely have been drawn from the normal distribution:
+
+```{r}
+data <- table$V4_Scale
+
+# Bin our original data (our histogram function did this for us above).
+observed <- 1:10
+for (i in 1:length(observed)) {
+  inds <- which((i - 1) <= data & data < i)
+  observed[i] <- length(inds)
+}
+
+density <- function(x) {
+  dnorm(x, mean = mu, sd = sigma)
+}
+
+# Get proportions of results by region of normal distribution.
+bins <- 1:10
+for (i in 1:length(bins)) {
+  bins[i] <- integrate(density, lower = (i - 1), upper = i)$value
+}
+
+# Calculate expected values.
+expected <- bins * sum(observed)
+chi_sq <- sum((observed - expected)^2 / expected)
+
+# We have ten bins and are measuring one variable, so DoF = 9
```

Comment:
@kingfi -- Can you confirm this? I still get tripped up by degrees of freedom.

Reply:
I went back through my notes and I'm pretty sure this is correct. You draw from one sample so you'd only need to factor that one sample into your df calculation.